nss-day-cohort-39/tabloid-full-stack-velours
# view all categories

### Server-Side Changes
- Added Category Model, Repository, and Controller 
- Sorted Category List alphabetically
- Added Category object to ApplicationDbContext 
- Removed cached Firebase Project Id

### Client-Side Changes
- Added HTML rendering for Category object in Category.js
- Mapped through Category objects to display in a grid in CategoryList.js
- Used reactstrap ListGroup to style Category List (https://reactstrap.github.io/components/listgroup/)
- Created CategoryProvider.js with children "categories" and "GetCategories"
- Added route to ApplicationViews.js
- Added CategoryProvider to App.js

### Additional Changes
- Deleted blue buttons from Welcome page
- Added a path to logout so that it refreshes to the welcome page (was staying on the same screen)
- Edited for capitalization continuity

### Testing
- Pull branch and confirm that your Firebase Product Id is correct before testing
- Execute/Debug in VS and `npm start` from /Client 
- Login to Tabloid
- Confirm that "Posts," "MyPosts," and "Categories" appear in header and that all routes still work
- Click "Categories" link in header
- Confirm that all categories are listed in ListGroup format with count bubbles to the right
- Confirm that categories are in alphabetical order
- Logout and confirm that page refreshes to Welcome page
